starlette/testclient.py

```diff
@@ -451,7 +472,16 @@ def __enter__(self) -> requests.Session:
 
     def __exit__(self, *args: typing.Any) -> None:
         loop = asyncio.get_event_loop()
-        loop.run_until_complete(self.wait_shutdown())
+        shutdown = asyncio.ensure_future(self.wait_shutdown())
+        while True:
+            try:
+                loop.run_until_complete(shutdown)
+            except BaseException as exc:
+                if exc is self.adapter.handled_server_exception:
+                    continue
+                raise
+            else:
+                break
```

Comment:
The `break` is reported as not covered in py38, but is - likely some optimization not handled by (current) coverage.py

Reply:
I consider this to be quite a hack anyway - but at least this is finally getting not in my way anymore - and has tests for the different variants.